extras/rya.pcj.fluo/pcj.fluo.app/src/main/java/org/apache/rya/indexing/pcj/fluo/app/query/StatementPatternIdCacheSupplier.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.rya.indexing.pcj.fluo.app.query;
+
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class StatementPatternIdCacheSupplier {
+
+    private static final Logger LOG = LoggerFactory.getLogger(StatementPatternIdCacheSupplier.class);
+    private static boolean initialized = false;
+    private static StatementPatternIdCache CACHE;
+    private static final ReentrantLock lock = new ReentrantLock();
+
+    /**
+     * Returns an existing cache if one has been created, otherwise creates a new cache.
+     *
+     * @return - existing StatementPatternIdCache or new cache if one didn't already exist
+     */
+    public static StatementPatternIdCache getOrCreateCache() {
+        lock.lock();
+        try {
+            if (!initialized) {
+                LOG.debug("Cache has not been initialized.  Initializing StatementPatternIdCache");
+                CACHE = new StatementPatternIdCache();
+                initialized = true;
+            } else {
+                LOG.debug("A StatementPatternIdCache has already been initialized.");
+            }
+            return CACHE;
+        } finally {
+            lock.unlock();
+        }
+    }
+
+    /**
+     * Flushes stored cache and flags Supplier as uninitialized.
```

Comment:
should this be renamed to flush() then?

Reply:
I think clear is more consistent with Java method names.  For example, clear() is the method used to delete the contents of hash tables, sets, etc.

Reply:
Changed flushes to deletes in the docs.